File: app/src/main/java/com/hippo/ehviewer/spider/SpiderQueen.java
/*
 * STUB (LANraragi: E-Hentai spider system removed)
 * LANraragi uses LRRGalleryProvider / LRRDownloadWorker instead.
 */

package com.hippo.ehviewer.spider;

import android.content.Context;
import androidx.annotation.IntDef;
import androidx.annotation.NonNull;
import androidx.annotation.UiThread;

import com.hippo.ehviewer.client.data.GalleryInfo;
import com.hippo.lib.glgallery.GalleryProvider;
import com.hippo.lib.image.Image;
import com.hippo.unifile.UniFile;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * SpiderQueen — STUB.
 * E-Hentai image downloading spider has been removed.
 * Only preserved: constants, OnSpiderListener interface, findStartPage().
 */
public final class SpiderQueen implements Runnable {

    public static final int MODE_READ = 0;
    public static final int MODE_DOWNLOAD = 1;
    public static final int STATE_NONE = 0;
    public static final int STATE_DOWNLOADING = 1;
    public static final int STATE_FINISHED = 2;
    public static final int STATE_FAILED = 3;
    public static final int DECODE_THREAD_NUM = 2;
    public static final String SPIDER_INFO_FILENAME = ".ehviewer";
    public static final String SPIDER_INFO_BACKUP_DIR = "backupDir";

    private SpiderQueen() {}

    @UiThread
    public static SpiderQueen obtainSpiderQueen(@NonNull Context context,
            @NonNull GalleryInfo galleryInfo, @Mode int mode) {
        return new SpiderQueen();
    }

    @UiThread
    public static void releaseSpiderQueen(@NonNull SpiderQueen queen, @Mode int mode) {
        // No-op
    }

    @UiThread
    public static int findStartPage(@NonNull Context context, @NonNull GalleryInfo galleryInfo) {
        return 0;
    }

    public void addOnSpiderListener(OnSpiderListener listener) {}
    public void removeOnSpiderListener(OnSpiderListener listener) {}

    public int size() { return GalleryProvider.STATE_ERROR; }
    public String getError() { return "E-Hentai spider removed"; }
    public Object request(int index) { return null; }
    public Object forceRequest(int index) { return null; }
    public void cancelRequest(int index) {}
    public int getStartPage() { return 0; }
    public void putStartPage(int page) {}

    public boolean save(int index, @NonNull UniFile file) { return false; }
    public UniFile save(int index, @NonNull UniFile dir, @NonNull String filename) { return null; }

    @Override
    public void run() {}

    @IntDef({MODE_READ, MODE_DOWNLOAD})
    @Retention(RetentionPolicy.SOURCE)
    public @interface Mode {}

    @IntDef({STATE_NONE, STATE_DOWNLOADING, STATE_FINISHED, STATE_FAILED})
    @Retention(RetentionPolicy.SOURCE)
    public @interface State {}

    public interface OnSpiderListener {
        void onGetPages(int pages);
        void onGet509(int index);
        void onPageDownload(int index, long contentLength, long receivedSize, int bytesRead);
        void onPageSuccess(int index, int finished, int downloaded, int total);
        void onPageFailure(int index, String error, int finished, int downloaded, int total);
        void onFinish(int finished, int downloaded, int total);
        void onGetImageSuccess(int index, Image image);
        void onGetImageFailure(int index, String error);
    }
}

File: app/src/main/java/com/hippo/ehviewer/spider/SpiderQueen.kt
/*
 * STUB (LANraragi: E-Hentai spider system removed)
 * LANraragi uses LRRGalleryProvider / LRRDownloadWorker instead.
 */

package com.hippo.ehviewer.spider

import androidx.annotation.IntDef
import com.hippo.lib.glgallery.GalleryProvider
import com.hippo.lib.image.Image
import com.hippo.unifile.UniFile

/**
 * SpiderQueen -- STUB.
 * E-Hentai image downloading spider has been removed. Only retained:
 *  - [Mode] / [State] annotations and constants
 *  - [OnSpiderListener] callback interface (used by LRRDownloadWorker /
 *    DownloadScheduler)
 *  - [SPIDER_INFO_FILENAME] / [SPIDER_INFO_BACKUP_DIR] constants
 *
 * The instance class is itself dead — no caller creates a SpiderQueen.
 * The class shell is preserved only because Kotlin nests its constants
 * and the listener interface inside it; removing the class would mean
 * relocating those declarations to a new top-level container.
 */
class SpiderQueen private constructor() : Runnable {

    fun addOnSpiderListener(listener: OnSpiderListener?) {}
    fun removeOnSpiderListener(listener: OnSpiderListener?) {}

    fun size(): Int = GalleryProvider.STATE_ERROR
    fun getError(): String = "E-Hentai spider removed"
    fun request(index: Int): Any? = null
    fun forceRequest(index: Int): Any? = null
    fun cancelRequest(index: Int) {}
    fun getStartPage(): Int = 0
    fun putStartPage(page: Int) {}

    fun save(index: Int, file: UniFile): Boolean = false
    fun save(index: Int, dir: UniFile, filename: String): UniFile? = null

    override fun run() {}

    @IntDef(MODE_READ, MODE_DOWNLOAD)
    @Retention(AnnotationRetention.SOURCE)
    annotation class Mode

    @IntDef(STATE_NONE, STATE_DOWNLOADING, STATE_FINISHED, STATE_FAILED)
    @Retention(AnnotationRetention.SOURCE)
    annotation class State

    interface OnSpiderListener {
        fun onGetPages(pages: Int)
        fun onGet509(index: Int)
        fun onPageDownload(index: Int, contentLength: Long, receivedSize: Long, bytesRead: Int)
        fun onPageSuccess(index: Int, finished: Int, downloaded: Int, total: Int)
        fun onPageFailure(index: Int, error: String, finished: Int, downloaded: Int, total: Int)
        fun onFinish(finished: Int, downloaded: Int, total: Int)
        fun onGetImageSuccess(index: Int, image: Image)
        fun onGetImageFailure(index: Int, error: String)
    }

    companion object {
        const val MODE_READ = 0
        const val MODE_DOWNLOAD = 1
        const val STATE_NONE = 0
        const val STATE_DOWNLOADING = 1
        const val STATE_FINISHED = 2
        const val STATE_FAILED = 3
        const val DECODE_THREAD_NUM = 2
        const val SPIDER_INFO_FILENAME = ".ehviewer"
        const val SPIDER_INFO_BACKUP_DIR = "backupDir"
    }
}
